### PR TITLE
Webpack cleanup

### PR DESCRIPTION
**Pull request by timkelty**
Lots of refactor here, so here's the summary:

- removed need for empty `entry.js` files
    - this also removed the need to `removeFiles`
- no more `lib`, `asset`, 'base' types…`vue` is the only remaining `type`
- Removed the `.min` output default
    - There was already lots of inconsistency here (eg css files), so it seemed better to just drop the `.min`. 

Note: There are instances where we're using `CopyWebpackPlugin` (eg jquery) where we could simply use the module as the entry. I've left this in place, because we need the globals exposed to the window object and this is alot more straightforward than dealing with `expose-loader`, `exports-loader`, `DefinePlugin`, etc.

### CODE REVIEW

File: packages/craftcms-webpack/index.js [hunk 1]
```diff
@@ -0,0 +1,11 @@
+const glob = require('glob');
+const path = require('path');
+const fetchConfigs = (
```
Comment (nfourtythree) [inline packages/craftcms-webpack/index.js:3]: Was it not possible to put this in its own file having the "base path" passed into the constructor? With there default being the `module.parent.path`?

**Review by brandonkelly — APPROVED** [review]

File: src/web/assets/jqueryui/webpack.config.js [hunk 2]
```diff
@@ -21,9 +20,6 @@ module.exports = new CraftWebpackConfig({
                         require.resolve('jquery-ui/ui/widgets/mouse.js'),
                     ],
                 },
-                transform: {
-                    'jquery-ui.js': code => require("uglify-js").minify(code).code
-                }
```
Comment (timkelty) [inline src/web/assets/jqueryui/webpack.config.js:26]: This is already getting minified, so this isn't needed.

File: lerna.json [hunk 3]
```diff
@@ -1,5 +1,10 @@
 {
   "packages": ["packages/*"],
   "npmClient": "npm",
-  "version": "independent"
+  "version": "independent",
+  "command": {
+    "version": {
+      "allowBranch": ["main", "develop"]
+    }
+  }
```
Comment (timkelty) [inline lerna.json:9]: So we don't accidentally release on a feature branch

File: packages/craftcms-webpack/get-config.js [hunk 4]
```diff
@@ -0,0 +1,309 @@
...
+};
+
+module.exports = ({
+  context,
+  type = null,
+  config = {},
+  watchPaths = [
+    path.join(rootPath, "src/templates"),
+    path.join(context, "dist"),
+  ],
```
Comment (timkelty) [inline packages/craftcms-webpack/get-config.js:140]: Replaced `templatePath` with `watchPaths`

File: packages/craftcms-webpack/get-config.js [hunk 5]
```diff
@@ -0,0 +1,309 @@
...
+    client: {
+      overlay: {
+        errors: true,
+        warnings: false,
+      },
+    },
+    static: watchPaths.map(directory => ({
+      directory,
+      watch: true,
+    })),
```
Comment (timkelty) [inline packages/craftcms-webpack/get-config.js:121]: @nfourtythree I killed the `DEV_SERVER_CONTENT_BASE` env var…do we need it? Seems like you could just pass `watchPaths` for whatever you wanted to watch.

File: packages/craftcms-webpack/index.js [hunk 6]
```diff
@@ -0,0 +1,347 @@
...
+ * Returns and array of webpack configs.
+ *
+ * @param {string} globPattern
+ * @param {Object} options
+ * @param {string} options.cwd Path where the glob should be executed from.
+ */
+const getConfigs = (
+  globPattern = "src/web/assets/*/webpack.config.js",
+  options = {
+    cwd: module.parent.path,
```
Comment (nfourtythree) [inline packages/craftcms-webpack/index.js:44]: @timkelty Should we rename this to be like the other changes you made?

```suggestion
    context: module.parent.path,
```

Comment (nfourtythree) [reply packages/craftcms-webpack/index.js:44]: This is not required because we are just passing the glob options straight through

**Review by nfourtythree — APPROVED** [review]

**Pull request merged**